CMakeLists.txt

```
@@ -609,13 +620,25 @@ endif()
 option(CURL_ZSTD "Set to ON to enable building curl with zstd support." OFF)
 set(HAVE_ZSTD OFF)
 if(CURL_ZSTD)
-  find_package(Zstd REQUIRED)
-  if(Zstd_FOUND AND NOT Zstd_VERSION VERSION_LESS "1.0.0")
-    set(HAVE_ZSTD ON)
-    list(APPEND CURL_LIBS ${Zstd_LIBRARIES})
-    include_directories(${Zstd_INCLUDE_DIRS})
+  hunter_add_package(zstd)
+  if(HUNTER_ENABLED)
+    find_package(zstd REQUIRED)
+    if(Zstd_FOUND AND NOT Zstd_VERSION VERSION_LESS "1.0.0")
+      set(HAVE_ZSTD ON)
+      list(APPEND CURL_LIBS zstd::libzstd_static)
```

This would break if you'd tell hunter to compile zstd as shared library, which a user could reasonably do

---

you could check if there is a static lib version and prefer that, and otherwise use the non-static target

```suggestion
      if(TARGET zstd::libzstd_static)
        list(APPEND CURL_LIBS zstd::libzstd_static)
      else()
        list(APPEND CURL_LIBS zstd::libzstd_shared)
      endif()
```